src/Build/BuildCheck/Infrastructure/CheckWrapper.cs

```diff
@@ -52,10 +68,33 @@ internal void StartNewProject(
         }
     }
 
+    internal void ReportResult(BuildCheckResult result, ICheckContext checkContext, CheckConfigurationEffective config)
+    {
+        if (_limitReportsNumber)
+        {
+            if (_reportsCount > MaxReportsNumberPerRule)
+            {
+                return;
+            }
+
+            if (_reportsCount == MaxReportsNumberPerRule)
+            {
+                checkContext.DispatchAsCommentFromText(MessageImportance.Normal, $"The check '{Check.FriendlyName}' has exceeded the maximum number of results allowed. Any additional results will not be displayed.");
+                _reportsCount++;
+                return;
+            }
```

Comment:
Let's detect on the last diagnostic we want to emit (so MaxReportsNumberPerRule - 1) - so that the created report is not created unnecessary.
This applies once the dismounting is added (so basically - detect this is last wanted report - report it, report the note that the check will be dismounted and dismount the check)

Reply:
I thought about it, but I decided that not writing this message when it is not needed (for example when we have exactly 20 messages) is also a possible good behavior. Without counting dismounting of the check it is better. If we implement dismounting, we will indeed write it before dismount which is better to do during the last message exactly as you stated.